### PR TITLE
rework madara theme and fix many parser

### PR DESCRIPTION
**Pull request by davvarrr**
on mangareader : 
Delete table mode for automatic detect mode and chage defaut value for date and list url and fix.
and add to load images that are not layzy

on madara :

I merged all the madara templates into 1 and simplified the parse creation, taking inspiration from what was done on the MangaReaderParser theme to separate the sites in each language.

list of fix :
manhwa kool -> no chapter
BibiManga -> nothing found
beautymanga -> remove ( web site down )
manhwaindo-> fix load page
KumaPoi  -> name and domain change
fr-scan -> parse state
japscan -> load page.
Manga Manhua -> wrong date paterne
zinmanga - wrong date paterne
ozulscans - wrong date paterne and state not parse - Fix
turktoon - state not parse
Tempestfansub - state not parse
KomikDewasa - state not parse
komik av - wrong date paterne
komikindo - state not parse
KomikTap - state not parse
Mangasusu - state not parse
Manhwaland - > ( the domain name has changed) -> komikgo - wrong date paterne
manhwalist - ( the domain name has changed)
kiryuu - state not parse
manhwadesu - wrong date paterne
Soul Scans - state not parse
duniakomik - state not parse
kanzenin - wrong date paterne
komikstation - wrong date paterne
komik mama - wrong date paterne
Komik Manhwa - state not parse
komiku - state not parse
mangakita - state not parse and re create file.
manga susuku - state not parse
masterkomik - wrong date paterne

### CODE REVIEW

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/JapScanParser.kt [hunk 1]
```diff
@@ -112,16 +112,21 @@ internal class JapScanParser(context: MangaLoaderContext) : PagedMangaParser(con
...
+		val longkeyRegex = Regex("""'([\dA-Z]{20})'""", RegexOption.IGNORE_CASE)
 
-		val keyTables = keyRegex.findAll(script)
-			.mapNotNullTo(ArrayList(2)) { match ->
-				match.groupValues[1].takeIf {
-					it.toList().sorted() == sample
-				}
-			}
-		check(keyTables.size == 2)
+		val LongTables = longkeyRegex.findAll(script).map {
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/JapScanParser.kt:118]:
```suggestion
		val longTables = longkeyRegex.findAll(script).map {
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/JapScanParser.kt [hunk 2]
```diff
@@ -112,16 +112,21 @@ internal class JapScanParser(context: MangaLoaderContext) : PagedMangaParser(con
...
+			it.groupValues[1]
+		}.toList()
+
+		val ShortTables = shortkeyRegex.findAll(script).map {
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/JapScanParser.kt:122]:
```suggestion
		val shortTables = shortkeyRegex.findAll(script).map {
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/MadaraParser.kt [hunk 3]
```diff
@@ -38,6 +38,28 @@ internal abstract class MadaraParser(
...
+
+	protected val ongoing: Array<String> = arrayOf(
+	"مستمرة", "En curso", "En Curso","Ongoing", "OnGoing","On going",
+	"Ativo", "En Cours", "En cours", "Đang tiến hành", "Em lançamento", "em lançamento", "Em Lançamento", "Онгоінг", "Publishing",
+	"Devam Ediyor", "Em Andamento", "In Corso", "Güncel", "Berjalan", "Продолжается", "Updating",
+	"Lançando", "In Arrivo", "Emision", "En emision", "مستمر", "Curso", "En marcha", "Publicandose", "连载中",)
+
+	protected val finished: Array<String> = arrayOf(
+	"Completed", "Completo", "Complété", "Fini", "Terminé", "Tamamlandı", "Đã hoàn thành", "مكتملة", "Завершено",
+	"Finished", "Finalizado", "Completata", "One-Shot", "Bitti", "Tamat", "Completado", "Concluído", "Concluido", "已完结",)
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/MadaraParser.kt:60]: Looks like this one is not formatted correctly

**Review by Koitharu — APPROVED** [review]

**Comment by Koitharu** [comment]
Thank you, great job

**Comment by davvarrr** [comment]
Thank you too ;) I prepare a lot of new sources based on the theme of Madara I would post it in the Wekends:)

**Comment by Koitharu** [comment]
Please use the _Reformat code_ action before commit to keep code style in consistent state
изображение

**Comment by davvarrr** [comment]
no worries I'll do that ;)

**Pull request merged**